AIO/src/org/aio/activities/quests/SheepShearer.java

```diff
@@ -0,0 +1,150 @@
+package org.aio.activities.quests;
+
+import org.aio.activities.activity.Activity;
+import org.aio.activities.banking.DepositAllBanking;
+import org.aio.util.MakeAllInterface;
+import org.aio.util.Sleep;
+import org.osbot.rs07.api.map.Area;
+import org.osbot.rs07.api.model.GroundItem;
+import org.osbot.rs07.api.model.NPC;
+import org.osbot.rs07.api.model.RS2Object;
+import org.osbot.rs07.api.ui.Tab;
+
+public class SheepShearer extends QuestActivity {
+
+    private MakeAllInterface makeAllInterface;
+
+    private static final Area FARMER = new Area(3188, 3275, 3190, 3270);
+    private static final Area SHEEP = new Area(3195, 3272, 3209, 3260);
+    private static final Area SPINNER = new Area(3209, 3215, 3211, 3212).setPlane(1);
+
+    private static final String[] FARMER_OPTIONS = {
+            "I'm looking for a quest.",
+            "Yes okay. I can do that.",
+            "Of course!",
+            "I'm something of an expert actually!"
+    };
+
+    private static final int INVENTORY_SLOTS_REQUIRED = 22;
+
+    private static final String[] ITEMS_NEEDED = {
+            "Ball of wool",
+            "Wool",
+            "Shears"
+    };
+
+    private final DepositAllBanking depositAllBanking = new DepositAllBanking(ITEMS_NEEDED);
+
+    public SheepShearer() {
+        super(Quest.SHHEP_SHEARER);
+    }
+
+    @Override
+    public void onStart() {
+        depositAllBanking.exchangeContext(getBot());
+
+        makeAllInterface = new MakeAllInterface("Ball of Wool");
+        makeAllInterface.exchangeContext(getBot());
+    }
+
+    @Override
+    public void runActivity() throws InterruptedException {
+        if (getInventory().getEmptySlotCount() < INVENTORY_SLOTS_REQUIRED - getInventory().getAmount(ITEMS_NEEDED)) {
+            depositAllBanking.run();
+        } else if (getTabs().getOpen() != Tab.INVENTORY) {
+            getTabs().open(Tab.INVENTORY);
+        } else {
+            switch (getProgress()) {
+                case 0:
+                    talkToFarmer();
+                    break;
+                case 1:
+                    if (hasRequiredItems()) {
+                        talkToFarmer();
+                    } else {
+                        getItemsNeeded();
+                    }
+                    break;
+                case 21:
+                    log("Quest is complete");
+                    isComplete = true;
+                    break;
+                default:
+                    log("Unknown progress config value: " + getProgress());
+                    setFailed();
+                    break;
+            }
+        }
+    }
+
+    private boolean hasRequiredItems() {
+        return getInventory().getAmount("Ball of wool") > 19;
+    }
+
+    private void getItemsNeeded() throws InterruptedException {
+        if (!getInventory().contains("Shears")) {
```

Comment:
I think @Explv might have more to say on this, but why use if/else instead of just 3 separate if statements?

That way a single call to this function ensures that by the end of its completion you have all of the required items.

Reply:
I would argue that it's better to have it run multiple times, so that if an interrupt call gets called upon it will break out of it faster?

Reply:
Yeah that’s fair. I don’t use it in my scripts but with short actions like this I don’t think it makes a difference. If it works it works!